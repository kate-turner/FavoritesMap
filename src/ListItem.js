import React from "react";
import {Button, List} from 'semantic-ui-react'
import PropTypes from "prop-types";

export default function ListItem(props) {
    const {handleSetActiveViewPort, favorite, handleUpdateFavoritesList } = props

    function onFavoriteClick(favorite) {
        handleSetActiveViewPort({
            center: [favorite.lng, favorite.lat],
            zoom: 15
        })
    }

    function onDeleteClick(favorite) {
        handleUpdateFavoritesList(favorite, "delete")
    }

    return (
        <List.Item>
            <List.Content>
                <List.Header as='a' onClick={() => onFavoriteClick(favorite)}> {favorite.text}
                </List.Header>

                <List.Description>
                    {favorite.place_name}
                </List.Description>
                <List.Description>
                    <Button onClick={() => onDeleteClick(favorite)}>Delete</Button>
                </List.Description>
            </List.Content>
        </List.Item>

    )
}

ListItem.propTypes = {
    favorite: PropTypes.object.isRequired,
    handleUpdateFavoritesList: PropTypes.func.isRequired,
    handleSetActiveViewPort: PropTypes.func.isRequired,
}

ListItem.defaultProps = {
    favorite: {},
    handleUpdateFavoritesList: () => {},
    handleSetActiveViewPort: () => {}
}