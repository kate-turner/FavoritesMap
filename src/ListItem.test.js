import React from 'react';
import {mount} from 'enzyme';
import {mountToJson} from "enzyme-to-json";
import ListItem from "./ListItem";

describe('ListItem', () => {
    let mockCallback = jest.fn()
    let wrapper;

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = mount(<ListItem handleUpdateFavoritesList={mockCallback} favorite={{}} handleSetActiveViewPort={mockCallback}/>)
    });

    it('renders list item', () => {
        expect(mountToJson(wrapper)).toMatchSnapshot();
    })

    it('it renders list item delete button', () => {
        let deleteButton = wrapper.find('button')
        expect(deleteButton).toHaveLength(1);
    })

    it('calls handleUpdateFavoritesList on delete click', () => {
        let deleteButton = wrapper.find('button')
        deleteButton.simulate('click')
        expect(mockCallback.mock.calls.length).toEqual(1);
    })

    it('renders clickable location name', () => {
        let locationName = wrapper.find('.header')
        expect(locationName).toHaveLength(1);
    })
    it('calls handleSetActiveViewPort on location name click', () => {
        let locationName = wrapper.find('.header')
        locationName.simulate('click')
        expect(mockCallback.mock.calls.length).toEqual(1);
    })
})