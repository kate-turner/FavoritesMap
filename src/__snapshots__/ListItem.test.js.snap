// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem renders list item 1`] = `
<ListItem
  favorite={Object {}}
  handleSetActiveViewPort={[MockFunction]}
  handleUpdateFavoritesList={[MockFunction]}
>
  <ListItem>
    <div
      className="item"
      onClick={[Function]}
      role="listitem"
    >
      <ListContent>
        <div
          className="content"
        >
          <ListHeader
            as="a"
            onClick={[Function]}
          >
            <a
              className="header"
              onClick={[Function]}
            >
               
              undefined/undefined
            </a>
          </ListHeader>
          <ListDescription>
            <div
              className="description"
            />
          </ListDescription>
          <ItemExtra>
            <div
              className="extra"
            >
              <Button
                as="button"
                onClick={[Function]}
                size="mini"
              >
                <Ref
                  innerRef={
                    Object {
                      "current": <button
                        class="ui mini button"
                      >
                        Delete
                      </button>,
                    }
                  }
                >
                  <RefFindNode
                    innerRef={
                      Object {
                        "current": <button
                          class="ui mini button"
                        >
                          Delete
                        </button>,
                      }
                    }
                  >
                    <button
                      className="ui mini button"
                      onClick={[Function]}
                    >
                      Delete
                    </button>
                  </RefFindNode>
                </Ref>
              </Button>
            </div>
          </ItemExtra>
        </div>
      </ListContent>
    </div>
  </ListItem>
</ListItem>
`;
